enterprise/backend/test/metabase_enterprise/util_test.clj

```diff
@@ -0,0 +1,56 @@
+(ns metabase-enterprise.util-test
+  (:require [metabase.public-settings.premium-features :refer [defenterprise]]
+            [schema.core :as s]))
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                          Defenterprise Macro                                                  |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+;; These `defenterprise` calls define the EE versions of test functions which are used in
+;; `metabase.public-settings.premium-features-test`, for testing the `defenterprise` macro itself
+
+(defenterprise greeting
+  "Returns an special greeting for anyone running the Enterprise Edition, regardless of token."
+  :feature :none
+  [username]
+  (format "Hi %s, you're running the Enterprise Edition of Metabase!" (name username)))
+
+(defenterprise greeting-with-schema :- s/Keyword
+  "Returns an special greeting for anyone running the Enterprise Edition, regardless of token."
+  :feature :none
+  [username]
+  {username s/Keyword}
+  (format "Hi %s, you're running the Enterprise Edition of Metabase!" (name username)))
```

Comment:
Built-in schema support is awesome 💪 but I don't think we _need_ to have it. If you do want to support it, it might be easier to have a separate `defenterprise-schema` macro or something that handles the schema stuff so the core `defenterprise` can focus on the other stuff. I'd also suggest trying to get the `[param :- Schema]` syntax supported by `schema.core/defn` working instead of the `{param Schema}` syntax used by `defendpoint` if you can. 

FWIW I'd consider dropping the schema stuff for now tho, if we need schema validation we can always do it in a different function

Reply:
I wouldn't try to handle schema stuff myself honestly I'd leverage `s/fn` to do that for me. Basically you can have your 

```clj
(defenterprise whatever :- Keyword [] ...)
```

transform to

```
(register-mapping!
  whatever
  my-namespace
  (s/fn whatever :- Keyword [] ...))
```

Reply:
Yeah, I added it because I wanted to migrate [the LDAP functions as a test case](https://github.com/metabase/metabase/pull/21687) and they both define schemas for the args & return values, and I didn't want to just remove them.

What's the advantage of having a separate macro for schemas? Just making the implementation of this one simpler?

Agree about the `:-` syntax. Went with the `{param Schema}` approach since I had a reference for how to do it in `defendpoint` but I definitely want to change this.

WDYT about keeping this for now so that we can start using the macro, and I'll try to change it to the `:-` syntax as a follow-up?

Reply:
Oh, hmm, just using `s/fn` probably makes the most sense.

Reply:
RE supporting schema: you don't need to do it as a separate version at all. But `s/fn` has some limitations, e.g. it doesn't support keyword varargs e.g. `& {:keys []}` or something like that. So you'd only want to use `s/fn` as opposed to `fn` if someone is specifying schemas (with `:-`), and that would require parsing the function tail which I would try really hard to avoid if I were you.

Here's an example of supporting schema with two different versions of the macro (and a shared `defenterprise-fn` macro they both use)

https://gist.github.com/camsaul/fa4f29bdc2a6aa6f92b2192d398639be#file-defenterprise_schema-clj